asciidoctorj-core/src/main/java/org/asciidoctor/cli/AsciidoctorInvoker.java

```diff
@@ -100,13 +103,37 @@ private void setVerboseLevel(AsciidoctorCliOptions asciidoctorCliOptions) {
     }
 
     private Asciidoctor buildAsciidoctorJInstance(AsciidoctorCliOptions asciidoctorCliOptions) {
-        Asciidoctor asciidoctor;
-        if(asciidoctorCliOptions.isLoadPaths()) {
-         asciidoctor = JRubyAsciidoctor.create(asciidoctorCliOptions.getLoadPaths());   
-        } else {
-            asciidoctor = JRubyAsciidoctor.create();
+        ClassLoader oldTccl = Thread.currentThread().getContextClassLoader();
+        try {
+            if (asciidoctorCliOptions.isClassPaths()) {
+                URLClassLoader tccl = createUrlClassLoader(asciidoctorCliOptions.getClassPaths());
+                Thread.currentThread().setContextClassLoader(tccl);
+            }
+            Asciidoctor asciidoctor;
+            if (asciidoctorCliOptions.isLoadPaths()) {
+                asciidoctor = JRubyAsciidoctor.create(asciidoctorCliOptions.getLoadPaths());
+            } else {
+                asciidoctor = JRubyAsciidoctor.create();
+            }
+            return asciidoctor;
+        } finally {
+            Thread.currentThread().setContextClassLoader(oldTccl);
+        }
+    }
+
+    private URLClassLoader createUrlClassLoader(List<String> classPaths) {
```

Comment:
> Because you can also invoke java -cp /path/to/dirFullOfJars/* my/MainClass.

Which AFAIK means it will look for *.class files. 

So if I specify `--cp /path/tp/myconverter.jar` will it still work?